x/config/configurator_test.go

```diff
@@ -834,6 +850,237 @@ func TestConfigurator(t *testing.T) {
 				return
 			},
 		},
+		{
```

Comment:
suggest a couple of negative tests:
 - fields without `interpolate` that contain interpolation strings like `${REDIS_PORT}` are not interpolated
 - map keys/values and slice values are not interpolated